platform-utils/endpoints-generator/index.js

```diff
@@ -0,0 +1,22 @@
+import endpoints from "./endpoints.js";
+import fs from "fs";
+
+const fallbackURL = "http://localhost:8000";
+const fallbackAuth = null;
+
+const endpointsWithFallback = Object.fromEntries(
+    Object.entries(endpoints).map(([key, endpoint]) => [
+        key,
+        {
+            url: endpoint.url || fallbackURL,
+            auth: endpoint.auth || fallbackAuth,
+        },
+    ])
+);
+
+fs.mkdirSync("../../assets/config");
+
+fs.writeFileSync(
+    "../../assets/config/endpoints.json",
```

Comment:
For extra safety, this should be outside the repository.

Reply:
The secrets will only be filled in in the github build right? So is that still relevant and does it work then?

Reply:
Um, still dangerous. They might show up in a log, and they'll be unencrypted in the filesystem. If they are being generated on the fly and committed to the filesystem... I don't know. Might want to check with @L-E, but my hunch is that you shouldn't do that. At any rate, you will probably have to create a very detailed data flow and data processing scheme to guarantee that nothing is leaked